x/auctions/keeper/msg_server.go

```diff
@@ -0,0 +1,108 @@
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/evmos/evmos/v18/utils"
+	"github.com/evmos/evmos/v18/x/auctions/types"
+	"github.com/pkg/errors"
+)
+
+var (
+	_ types.MsgServer = &Keeper{}
+)
+
+// Bid defines a method for placing a bid on an auction
+func (k Keeper) Bid(goCtx context.Context, bid *types.MsgBid) (*types.MsgBidResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params := k.GetParams(ctx)
+	if !params.EnableAuction {
+		return nil, errorsmod.Wrapf(types.ErrAuctionDisabled, "auction is disabled")
+	}
+
+	lastBid := k.GetHighestBid(ctx)
+	if bid.Amount.Amount.LT(lastBid.Amount.Amount) {
+		return nil, errors.Wrapf(types.ErrBidMustBeHigherThanCurrent, "bid amount %s is less than last bid %s", bid.Amount, lastBid.Amount)
+	}
+
+	senderAddr, err := sdk.AccAddressFromBech32(bid.Sender)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid sender address")
+	}
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, senderAddr, types.ModuleName, sdk.NewCoins(bid.Amount)); err != nil {
+		return nil, errors.Wrap(err, "deposit failed")
+	}
+
+	// If there is a last bid, refund it
+	if !lastBid.Amount.IsZero() {
+		if err := k.refundLastBid(ctx); err != nil {
+			return nil, errors.Wrap(err, "refund failed")
+		}
+	}
+	k.SetHighestBid(ctx, bid.Sender, bid.Amount)
+
+	// TODO: emit events
+
+	return &types.MsgBidResponse{}, nil
+}
+
+// DepositCoin defines a method for depositing coins into the auction module
+func (k Keeper) DepositCoin(goCtx context.Context, deposit *types.MsgDepositCoin) (*types.MsgDepositCoinResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := k.GetParams(ctx)
+	if !params.EnableAuction {
+		return nil, errorsmod.Wrapf(types.ErrAuctionDisabled, "auction is disabled")
+	}
+
+	senderAddr, err := sdk.AccAddressFromBech32(deposit.Sender)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid sender address")
+	}
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, senderAddr, types.AuctionCollectorName, sdk.NewCoins(deposit.Amount)); err != nil {
+		return nil, errors.Wrap(err, "deposit failed")
+	}
+
+	// TODO: Emit Events
+
+	return &types.MsgDepositCoinResponse{}, nil
+}
+
+// UpdateParams defines a method for updating inflation params
+func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if k.authority.String() != req.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority.String(), req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := k.SetParams(ctx, req.Params); err != nil {
+		return nil, errorsmod.Wrapf(err, "error setting params")
+	}
+
+	return &types.MsgUpdateParamsResponse{}, nil
+}
+
+// refundLastBid refunds the last bid placed on an auction
+func (k Keeper) refundLastBid(ctx sdk.Context) error {
+
+	previousBid := k.GetHighestBid(ctx)
+	previousBidAmount := previousBid.Amount.Amount
```

Comment:
nit: consistency with function name and other vars:
```suggestion
	lastBid := k.GetHighestBid(ctx)
	lastBidAmount := previousBid.Amount.Amount
```